Source/DafnyLanguageServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -772,18 +772,29 @@ public class DafnyModel {
           return result;
         }
 
-        // Otherwise, sequences can be reconstructed index by index,
-        // ensuring indices are in ascending order.
-        IEnumerable<(Model.Element, BigInteger)> tpls =
-          fSeqIndex.AppsWithArg(0, var.Element)
-          .Select(tpl => (Unbox(tpl.Result), BigInteger.Parse(tpl.Args[1].ToString())))
-          .OrderBy(tpl => tpl.Item2);
-
-        foreach (var (res, idx) in tpls) {
-          var idxstr = idx.ToString();
-          var e = DafnyModelVariableFactory.Get(state, res, "[" + idxstr + "]", var);
+        // Otherwise, sequences can be reconstructed index by index, ensuring indices are in ascending order.
+        // NB: per https://github.com/dafny-lang/dafny/issues/3048 , not all indices may be parsed as a BigInteger,
```

Comment:
If your file with the reproducing example is MIT-licensed, would you mind adding it as a test case in DafnyLanguageServer.Text/Various/TestFiles and ensure this PR is tested? I think it's an excellent file that would be good to keep around, even if it is big.

Reply:
Sure, I was going to do that but wondered if it was too long to do so, so I never followed through - will do.

Reply:
Hi @MikaelMayer , the latest commit includes the repro case as a unit test; let me know if isn't quite wired up in the way you'd like.